### PR TITLE
Fix some bugs

### PR DESCRIPTION
**Pull request by VoicilaIonut**
Added FILE* in task_details.
Closed and opened fd at pause and resume.
Added set_task_status as a driver when we want to change the status of a task [still can be improved]
Tested with resume, paused, remove (only when the task is finished)
Please leave syslog as a debugging method until we have a stable version

### CODE REVIEW

File: daemon/include/task.h [hunk 1]
```diff
@@ -2,14 +2,11 @@
 #define TASK_H
 
 #include <constants.h>
-
 #include <pthread.h>
 #include <shared.h>
 #include <stdio.h>
 #include <stdlib.h>
 
-#define MAX_OUTPUT_SIZE 16384
```
Comment (VoicilaIonut) [inline daemon/include/task.h:11]: moved in constants

File: daemon/include/task.h [hunk 2]
```diff
@@ -30,12 +28,7 @@ void output_task(struct task_details *task);
 void permission_to_continue(struct task_details *task);
 
 /*
- * Change task status.
- */
-void set_task_status(struct task_details *task, enum Status status);
```
Comment (VoicilaIonut) [inline daemon/include/task.h:35]: refactored

File: daemon/include/utils.h [hunk 3]
```diff
@@ -47,9 +47,10 @@ char *relative_path(const char *path, int depth);
  */
 int special_directory(char *d_name);
 
-void syslog_message(const struct message *msg);
-
-int get_depth(const char *path, const char *subpath);
+/*
+ * Returns the depth of the subpath in the path or -1 if the subpath is not in the path.
+ */
+int get_depth_of_subpath(const char *path, const char *subpath);
```
Comment (VoicilaIonut) [inline daemon/include/utils.h:53]: just renamed

File: daemon/include/utils.h [hunk 4]
```diff
@@ -59,4 +60,9 @@ void send_error_response(int client_fd, enum ResponseCode);
 
 char *status_to_string(enum Status status);
 
+/*
+ * Check if the directory exists, returns 1 if exits.
+ */
+int directory_exists(const char *path);
```
Comment (VoicilaIonut) [inline daemon/include/utils.h:66]: moved

File: daemon/main.c [hunk 5]
```diff
@@ -334,7 +294,15 @@ int main(void)
             }
 
             case LIST:
-                // TODO:
+                // TODO
+                for (int i = 0; i < MAX_TASKS; i++)
```
Comment (VoicilaIonut) [inline daemon/main.c:298]: please leave as a method to see the tasks we have in daemon

File: daemon/src/analyzer.c [hunk 6]
```diff
@@ -61,8 +53,9 @@ long long get_size_dir(const char *path, struct task_details *task)
     return size;
 }
 
-long long analyzing(const char *path, struct task_details *task, FILE *output_fd)
+long long analyzing(const char *path, struct task_details *task)
```
Comment (VoicilaIonut) [inline daemon/src/analyzer.c:56]: we store fd in the task now

File: daemon/src/analyzer.c [hunk 7]
```diff
@@ -111,55 +104,30 @@ void *start_analyses_thread(void *arg)
     current_task->total_size = get_size_dir(current_task->path, current_task);
     check_or_exit_thread(current_task->total_size != 0, current_task, "Total size = 0");
 
-    char thread_output[MAX_PATH_SIZE];
```
Comment (VoicilaIonut) [inline daemon/src/analyzer.c:114]: created a function for this to be called in init_task and in the resume_task

File: daemon/src/analyzer.c [hunk 8]
```diff
@@ -111,55 +104,30 @@ void *start_analyses_thread(void *arg)
...
-    snprintf(thread_output, MAX_PATH_SIZE, "/var/run/user/%d/da_tasks/tree_%d", getuid(), current_task->task_id);
```
Comment (VoicilaIonut) [inline daemon/src/analyzer.c:115]: the path is now in constants.h

File: daemon/src/analyzer.c [hunk 9]
```diff
@@ -111,55 +104,30 @@ void *start_analyses_thread(void *arg)
...
-    FILE *output_fd = fopen(thread_output, "w");
-    check_or_exit_thread(output_fd != NULL, current_task, "Error opening output file");
+    sleep(5);
```
Comment (VoicilaIonut) [inline daemon/src/analyzer.c:107]: added to be able to pause tasks for easly

File: daemon/src/analyzer.c [hunk 10]
```diff
@@ -111,55 +104,30 @@ void *start_analyses_thread(void *arg)
...
-    long long analyzing_size = analyzing(current_task->path, current_task, output_fd);
-    fclose(output_fd);
+    long long analyzing_size = analyzing(current_task->path, current_task);
     check_or_exit_thread(analyzing_size == current_task->total_size, current_task, "Different sizes");
+
     set_task_status(current_task, FINISHED);
     return NULL;
 }
 
-int directory_exists(const char *path)
```
Comment (VoicilaIonut) [inline daemon/src/analyzer.c:126]: moved

File: daemon/src/analyzer.c [hunk 11]
```diff
@@ -111,55 +104,30 @@ void *start_analyses_thread(void *arg)
...
-            exit(1); // An error occurred
-        }
-    }
-
-    // Check if the path is a directory
-    return S_ISDIR(statbuf.st_mode);
-}
-
-void write_report_info(FILE *output_fd, const char *path, long long size, struct task_details *task)
+void write_report_info(const char *path, long long size, struct task_details *task)
```
Comment (VoicilaIonut) [inline daemon/src/analyzer.c:116]: use the fd from the task details

File: daemon/src/task.c [hunk 12]
```diff
@@ -1,14 +1,44 @@
+
+
+#include <errno.h>
 #include <stdlib.h>
 #include <string.h>
 #include <syslog.h>
 #include <task.h>
+#include <utils.h>
+
+void output_task(struct task_details *task)
```
Comment (VoicilaIonut) [inline daemon/src/task.c:10]: for debugging, to be removed when we have a stable version

**Comment by oxygen-consumer** [comment]
>  Please leave syslog as a debugging method until we have a stable version

I would suggest to have an debug flag when compiling to run debug code only when is set, I can implement it. Please let me know if I should implement it.

Also this way we can leave the code used for debugging there even if we have a release version and just not setting the flag.

**Comment by VoicilaIonut** [comment]
> > Please leave syslog as a debugging method until we have a stable version
> 
> I would suggest to have an debug flag when compiling to run debug code only when is set, I can implement it. Please let me know if I should implement it.
> 
> Also this way we can leave the code used for debugging there even if we have a release version and just not setting the flag.

I think we should focus on MVP now and add the flag after.

**Review by beingsebi — APPROVED** [review]
gud

**Pull request merged**